db/migrate/20150227185631_add_users_state_flag.rb

```diff
@@ -0,0 +1,20 @@
+class AddUsersStateFlag < ActiveRecord::Migration
+  def up
+    add_column :users, :state, :string, default: 'temp'
+    execute "update users set state='temp' where is_temp='t'"
+    execute "update users set state='activated' where is_temp='f'"
+    # SQLite lacks support for the alter column statement
+    unless "SQLite" == ActiveRecord::Base.connection.adapter_name
+      execute "alter table users alter column state set not null"
+    end
+    remove_column :users, :is_temp
```

Comment:
@Dantemss That's about the best we can do to enforce non-null.  You can't use `null: false` in the add_column statement if the table has existing rows.  If you do so, it fails because the newly created column will start out as null. SQLite will allow you to, but PG will fail with: `ERROR:  column "state" contains null values`

PG lets you alter a column to set it to not null later though, which is what I'm doing here after it's been updated to have a value.